forest/jasmine/traj2stats.py

```diff
@@ -819,121 +862,248 @@ def gps_quality_check(study_folder: str, study_id: str) -> float:
+        person_point_radius: float, radius of the person's circle when
+            discovering places near him in pauses
+        place_point_radius: float, radius of place's circle
+            when place is returned as centre coordinates from osm
+        time_start: list, starting time of window of interest
+        time_end: list ending time of the window of interest
+            time should be a list of integers with format
+            [year, month, day, hour, minute, second]
+            if time_start is None and time_end is None: then it reads all
+            the available files
+            if time_start is None and time_end is given, then it reads all
+            the files before the given time
+            if time_start is given and time_end is None, then it reads all
+            the files after the given time
+        beiwe_study_ids: a list of beiwe IDs
+        parameters: Hyperparameters, hyperparameters in functions
+            recommend to set it to default
+        all_memory_dict: dict, from previous run (none if it's the first time)
+        all_bv_set: dict, from previous run (none if it's the first time)
+        quality_threshold: float, a percentage value of the fraction of data
+            required for a summary to be created.
+    Returns:
+        write summary stats as csv for each user during the specified
+            period
+        and a log of all locations visited as a json file if required
+        and imputed trajectory if required
+        and memory objects (all_memory_dict and all_bv_set)
+            as pickle files for future use
+        and a record csv file to show which users are processed
+        and logger csv file to show warnings and bugs during the run
     """
-    
-    quality_threshold = quality_threshold if quality_threshold is not None else 0.05
-    
-    if os.path.exists(output_folder)==False:
-        os.mkdir(output_folder)
-
-    if parameters == None:
-        parameters = [60*60*24*10,60*60*24*30,0.002,200,5,1,0.3,0.2,0.5,100,0.01,0.05,3,10,2,'GLC',10,51,None,None,None]
-    [l1,l2,l3,g,a1,a2,b1,b2,b3,d,sigma2,tol,switch,num,linearity,method,itrvl,accuracylim,r,w,h] = parameters
-    pars0 = [l1,l2,l3,a1,a2,b1,b2,b3]
-    pars1 = [l1,l2,a1,a2,b1,b2,b3,g]
-
-    if r == None:
+
+    os.makedirs(output_folder, exist_ok=True)
+
+    if quality_threshold is None:
+        quality_threshold = 0.05
+
+    pars0 = [
+        parameters.l1, parameters.l2, parameters.l3, parameters.a1,
+        parameters.a2, parameters.b1, parameters.b2, parameters.b3
+    ]
+    pars1 = [
+        parameters.l1, parameters.l2, parameters.a1, parameters.a2,
+        parameters.b1, parameters.b2, parameters.b3, parameters.g
+    ]
+
+    orig_r = parameters.r
+    orig_w = parameters.w
+    orig_h = parameters.h
+    if parameters.r is None:
         orig_r = None
-    if w == None:
+    if parameters.w is None:
         orig_w = None
-    if h == None:
+    if parameters.h is None:
```

Comment:
I think the three above conditionals are not necessary given the three preceding assignments?